### PR TITLE
Trip List (CA-71)

### PR DESCRIPTION
**Pull request by kcode20**
## Related Issues
https://caravanning.atlassian.net/browse/CA-71

## Description
In the dashboard component, create a trip list

###### DoD:
- Call GET /trips/:userid
- For each trip render the summary card component with the trip name and host.
- Pass the trip JSON to the Trip Detail Modal.

## Testing
Add a step-by-step list on how to test this PR, like this:
- [ ] Run the server by running `node server.js`
- [ ] Run the React Client by running `yarn start` or `npm start`
- [ ] Navigate to `localhost:3000/dashboard`. You should see the Trips Joined and Your Trips section.
Screen Shot 2020-04-04 at 10 12 07 PM

## How are you feeling?
😤

### CODE REVIEW

File: src/components/Dashboard/Dashboard.test.jsx [hunk 1]
```diff
@@ -2,9 +2,21 @@ import React from 'react';
 import { shallow } from 'enzyme';
 import Dashboard from './Dashboard';
 
+const mockTrip = {
```
Comment (jhrbva) [inline src/components/Dashboard/Dashboard.test.jsx:5]: should we remove this mockTrip now that other trips are rendering?

never mind.. just saw this was in test file, sorry

**Comment by jhrbva** [comment]
I just though of something: is this displaying all of the trips, even if accepted = false?
I feel like we should have a check so that if accepted is false the trip is not displayed. Unless we want the rejected trips to be displayed then I think they should go into another row.

**Comment by Quetourah** [comment]
> I just though of something: is this displaying all of the trips, even if accepted = false?
> I feel like we should have a check so that if accepted is false the trip is not displayed. Unless we want the rejected trips to be displayed then I think they should go into another row.

ah. yes. We should edit the query to filter out the trips that were rejected so they are not accepted. I don't know if you can do this in the frontend so we don't mess with the endpoint.

**Comment by kcode20** [comment]
@jhrbva @Quetourah if a user selects false when they are invited to a trip, they will never become a member of the trip (a column in the `members` table will NOT be created with their information) therefore this list will not have trips they did not accept. However, the filtering that you are talking about will need to be done on the endpoint for invitations. We will not want to show invitations that were already responded to.

**Comment by jhrbva** [comment]
> @jhrbva @Quetourah if a user selects false when they are invited to a trip, they will never become a member of the trip (a column in the `members` table will NOT be created with their information) therefore this list will not have trips they did not accept. However, the filtering that you are talking about will need to be done on the endpoint for invitations. We will not want to show invitations that were already responded to.

Should we create a ticket for the filtering on the backend?

**Pull request merged**